internal/controllers/config_manager.go

```diff
@@ -107,7 +112,7 @@ func (c *KubeEnvoyConfigManager) UpdateConfiguration(ctx context.Context, fleetI
 			return fmt.Errorf("failed to validate options: %w", err)
 		}
 
-		if err = UpdateConfigFromAPIOpts(envoyConfig, c.Validator, opts, apiSpec); err != nil {
+		if err = UpdateConfigFromAPIOpts(envoyConfig, c.Validator, opts, apiSpec, httpConnectionManagerBuilder); err != nil {
```

Comment:
I'm not sure if this is going to break anything else?
Maybe there is another way of injecting this. How can we detect that ext_authz filter is required before we do this here. If i'm not mistaken you only need httpConnectionManagerBuilder for this 
```	httpConnectionManagerBuilder.AppendFilterHTTPExternalAuthorizationFilterToStart(httpExternalAuthorizationFilter)
```